lib/src/controls/form/number_box.dart

```diff
@@ -0,0 +1,412 @@
+      _hasPrimaryFocus = focusNode!.hasPrimaryFocus;
+      focusNode!.addListener(_handleFocusChanged);
+    }
+
+    if (oldWidget.value != widget.value) {
+      if (widget.value != null) {
+        _updateController(widget.value!);
+      } else {
+        controller.text = '';
+      }
+    }
+  }
+
+  void _insertOverlay() {
+    _entry = OverlayEntry(builder: (context) {
+      assert(debugCheckHasMediaQuery(context));
+
+      final boxContext = _textBoxKey.currentContext;
+      if (boxContext == null) return const SizedBox.shrink();
+      final box = boxContext.findRenderObject() as RenderBox;
+
+      Widget child = PositionedDirectional(
+        width: kNumberBoxOverlayWidth,
+        child: CompositedTransformFollower(
+          link: _layerLink,
+          showWhenUnlinked: false,
+          offset: Offset(box.size.width - kNumberBoxOverlayWidth,
+              box.size.height / 2 - kNumberBoxOverlayHeight / 2),
+          child: SizedBox(
+            width: kNumberBoxOverlayWidth,
+            child: FluentTheme(
+              data: FluentTheme.of(context),
+              child: TextFieldTapRegion(
+                child: _NumberBoxCompactOverlay(
+                  onIncrement: _incrementSmall,
+                  onDecrement: _decrementSmall,
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+
+      return child;
+    });
+
+    if (_textBoxKey.currentContext != null) {
+      Overlay.of(context).insert(_entry!);
+      if (mounted) setState(() {});
+    }
+  }
+
+  void _dismissOverlay() {
+    _entry?.remove();
+    _entry = null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final textFieldSuffix = <Widget>[
+      if (widget.clearButton && _hasPrimaryFocus)
+        IconButton(
+          icon: const Icon(FluentIcons.clear),
+          onPressed: _clearValue,
+        ),
+    ];
+
+    switch (widget.mode) {
+      case SpinButtonPlacementMode.inline:
+        textFieldSuffix.addAll([
+          IconButton(
+            icon: const Icon(FluentIcons.chevron_up),
+            onPressed: widget.onChanged != null ? _incrementSmall : null,
+          ),
+          IconButton(
+            icon: const Icon(FluentIcons.chevron_down),
+            onPressed: widget.onChanged != null ? _decrementSmall : null,
+          ),
+        ]);
+        break;
+      case SpinButtonPlacementMode.compact:
+        textFieldSuffix.add(const SizedBox(width: kNumberBoxOverlayWidth));
+        break;
+      case SpinButtonPlacementMode.none:
+        break;
+    }
+
+    final child = TextBox(
+      key: _textBoxKey,
+      focusNode: focusNode,
+      controller: controller,
+      enabled: widget.onChanged != null,
+      suffix:
+          textFieldSuffix.isNotEmpty ? Row(children: textFieldSuffix) : null,
+    );
+
+    return CompositedTransformTarget(
+      link: _layerLink,
+      child: Focus(
+        onKeyEvent: (node, event) {
+          if (event is! KeyDownEvent) {
```

Comment:
`KeyRepeatEvent` should be allowed. This allows to keep the key pressed and keep incrementing